examples/string-templates/string_templates.bal

```diff
@@ -0,0 +1,21 @@
+import ballerina/io;
+
+public function main() {
+    // Note first \n is treated as is.
+    string s1 = string `line one \n rest of line 1 ${"\n"} second line`;
+    io:println(s1);
+
+    // You can use interpolations to add ` and  $ characters.
```

Comment:
Need to emphasize the characters using backticks. Need to check if/how this can be done.

Reply:
As far as I know we can't use markdown-ish formatting in side source code, they are rendered as is ([example](https://ballerina.io/learn/by-example/open-records/)). I didn't add backticks (`` ` ``)around since it would be confusing to the user (In markdown I believe you have wrap ` in side double backticks ``)